### PR TITLE
Add db-migrate tool and document basic migration strategy

### LINKED ISSUE #232: Implement DB migrations
Allow for automatic running of scripts for both UP and DOWN migrations.

As a matter of convention, I think we should follow a strategy used by Sequelize CLI where migration files are stored in the file system with a sortable name. Each migration is run in order, and the DB stores a list of executed migrations so that they are only run once per db instance.

### PR DESCRIPTION
**Pull request by DesignByOnyx**
This needs to be tested in PROD.

Closes #232

### CODE REVIEW

**Comment by justinbmeyer** [comment]
As an example, can you show adding a `startRate` and `endRate` to every `contributionMonth` from #231

File: migrations/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+```js
+const app = require('../src/app');
+
+exports.up = function() {
+  // return a Promise
+  return app.service('/api/os_projects').create({name: "CanJS"});
+}
+```
+
+**Exectute all new migration scripts:**
```
Comment (Alfredo-Delgado) [inline migrations/README.md]: typo

File: migrations/20170109181647-start-rate-end-rate-fields.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+exports.up = function(db) {
+	const svc = app.service('/api/contribution_months');
+	const ContributionMonth = svc.Model;
+	const paths = ContributionMonth.schema.paths;
+
+	/**
+	 * Example update direct to db
+	 */
+	return new Promise((resolve, reject) => {
```
Comment (justinbmeyer) [inline migrations/20170109181647-start-rate-end-rate-fields.js]:
```js
var update = Q.denodify( ContributionMonth, "update");


return update({
  startRate: { '$exists': false }
}, {
		startRate: paths.startRate.defaultValue,
		endRate: paths.endRate.defaultValue
});
```

Comment (justinbmeyer) [inline migrations/20170109181647-start-rate-end-rate-fields.js]: maybe use knex?

**Pull request merged**